skimage/filters/thresholding.py

```diff
@@ -203,29 +203,37 @@ def threshold_local(image, block_size, method='gaussian', offset=0,
     >>> binary_image2 = image > threshold_local(image, 15, 'generic',
     ...                                         param=func)
     """
-    if block_size % 2 == 0:
-        raise ValueError("The kwarg ``block_size`` must be odd! Given "
-                         "``block_size`` {0} is even.".format(block_size))
-    check_nD(image, 2)
+    if np.isscalar(block_size):
+        block_size = (block_size,) * image.ndim
+    elif len(block_size) != image.ndim:
+        raise ValueError("len(block_size) must equal image.ndim.")
+    block_size = tuple(block_size)
+    if any(b % 2 == 0 for b in block_size):
+        raise ValueError("block_size must be odd! Given block_size"
+                         "{0} contains even values.".format(block_size))
     thresh_image = np.zeros(image.shape, 'double')
     if method == 'generic':
         ndi.generic_filter(image, param, block_size,
                            output=thresh_image, mode=mode, cval=cval)
     elif method == 'gaussian':
         if param is None:
             # automatically determine sigma which covers > 99% of distribution
-            sigma = (block_size - 1) / 6.0
+            sigma = tuple([(b - 1) / 6.0 for b in block_size])
         else:
             sigma = param
         ndi.gaussian_filter(image, sigma, output=thresh_image, mode=mode,
                             cval=cval)
     elif method == 'mean':
-        mask = 1. / block_size * np.ones((block_size,))
-        # separation of filters to speedup convolution
-        ndi.convolve1d(image, mask, axis=0, output=thresh_image, mode=mode,
-                       cval=cval)
-        ndi.convolve1d(thresh_image, mask, axis=1, output=thresh_image,
-                       mode=mode, cval=cval)
+        masks = {}
+        for b in set(block_size):
+            masks[b] = 1. / b * np.ones((b,))
+        # # separation of filters to speedup convolution
+        b = block_size[0]
+        ndi.convolve1d(
+            image, masks[b], axis=0, output=thresh_image, mode=mode, cval=cval)
+        for ax, b in zip(range(1, image.ndim), block_size[1:]):
+            ndi.convolve1d(thresh_image, masks[b], axis=ax,
+                           output=thresh_image, mode=mode, cval=cval)
```

Comment:
```suggestion
        kernels = {b: np.full(b, 1/b) for b in set(block_size)}
        # separation of filters to speedup convolution
        thresh_image = image.copy()
        for ax, size in enumerate(block_size):
            ndi.convolve1d(thresh_image, kernels[b], axis=ax,
                           output=thresh_image, mode=mode, cval=cval)
```
Can't we here directly use the [`ndi.uniform_filter`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.uniform_filter.html) that is also implemented using the kernel separation?

Reply:
yes, I we can just make a single call to `uniform_filter`. thanks